### PR TITLE
DEVDOCS-4641: [new] Tax Provider and Connection API, added profile option

### PR DESCRIPTION
**Pull request by bc-andreadao**
# [DEVDOCS-4641]

Add info about Tax Profile option

[DEVDOCS-4641]: https://bigcommercecloud.atlassian.net/browse/DEVDOCS-4641?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/api-docs/provider-apis/tax/overview.mdx [hunk 1]
```diff
@@ -30,20 +30,35 @@ Once your tax provider configuration is ready, we'll let you know via email. The
...
-| Links displayed in an active MSF-enabled control panel | Optional | URL, Title, Description | Link(s) displayed to shoppers when they navigate to **Settings > Setup > Tax > _TaxProviderName_**. One or multiple links supported. | `support.sampletax.com`, `Sample Tax Support`, `The Sample Tax Support website.` |
+| Links displayed in an active MSF-enabled control panel | Optional | URL, Title, Description | Link(s) displayed to merchants when they navigate to **Settings > Setup > Tax > _TaxProviderName_**. One or multiple links supported. | `support.sampletax.com`, `Sample Tax Support`, `The Sample Tax Support website.` |
 **Coverage** | | | | |
-| Tax provider visibility | Required | Show, Hide | Tells us if tax provider should be displayed to users in the BigCommerce control panel before the tax providers app is installed on the associated store. | `Show` |
+| Tax provider visibility | Required | Show, Hide | Tells us if tax provider should be displayed to merchants in the BigCommerce control panel before the tax providers app is installed on the associated store. | `Show` |
```
Comment (bc-andreadao) [inline docs/api-docs/provider-apis/tax/overview.mdx:35]: Minor question, are the users merchants or **both** merchants and partners?

File: docs/api-docs/provider-apis/tax/overview.mdx [hunk 2]
```diff
@@ -30,20 +30,35 @@ Once your tax provider configuration is ready, we'll let you know via email. The
...
 | Tax provider type | Required | Production, Sandbox | Hierarchy of tax provider configurations, Production is primary and Sandboxes are secondary, see [sandbox tax provider configuration](#sandbox-tax-provider-configuration) for more information. | `Production` |
 | Partner support email | Required | Email | Used by BigCommerce to contact tax provider, in the case we need to forward a merchant support request. | `support@sampletax.example.com` |
-| Links displayed in an active MSF-enabled control panel | Optional | URL, Title, Description | Link(s) displayed to shoppers when they navigate to **Settings > Setup > Tax > _TaxProviderName_**. One or multiple links supported. | `support.sampletax.com`, `Sample Tax Support`, `The Sample Tax Support website.` |
+| Links displayed in an active MSF-enabled control panel | Optional | URL, Title, Description | Link(s) displayed to merchants when they navigate to **Settings > Setup > Tax > _TaxProviderName_**. One or multiple links supported. | `support.sampletax.com`, `Sample Tax Support`, `The Sample Tax Support website.` |
```
Comment (bc-andreadao) [inline docs/api-docs/provider-apis/tax/overview.mdx:33]: Minor question, should this be changed to merchants or **both** merchants and partners? (i.e. do partner stores have the same UI as merchant stores?)

File: docs/api-docs/provider-apis/tax/overview.mdx [hunk 3]
```diff
@@ -104,9 +119,9 @@ The [Get a Connection](/docs/apps-api/tax-app-connection#get-connection-status)
 
 ## Enabling tax providers in the control panel
 
-Once the tax provider's app has been successfully installed and basic authentication credentials have been provided via the Update a Connection request, users will be able to enable the tax provider on all supplied merchant and/or partner test stores provided by the tax provider in [sharing provider details with BigCommerce](#sharing-provider-details-with-bigcommerce).
+Once the tax provider's app has been successfully installed and basic authentication credentials have been provided via the Update a Connection request, merchants and partners will be able to enable the tax provider on all supplied merchant and/or partner test stores provided by the tax provider in [sharing provider details with BigCommerce](#sharing-provider-details-with-bigcommerce).
 
-To enable the tax provider, users must navigate to **Settings > Setup > Tax** in an active MSF-enabled BigCommerce control panel and click **Enable** next to the associated tax provider.
+To enable the tax provider, merchants must navigate to **Settings > Setup > Tax** in an active MSF-enabled BigCommerce control panel and click **Enable** next to the associated tax provider.
```
Comment (bc-andreadao) [inline docs/api-docs/provider-apis/tax/overview.mdx:124]: Minor question, should this be changed to merchants or **both** merchants and partners? (i.e. do partner stores have the same UI as merchant stores?)

File: docs/api-docs/provider-apis/tax/overview.mdx [hunk 4]
```diff
@@ -132,7 +147,7 @@ Estimate requests are not expected after the following events.
 Estimate calls will also be made in the control panel when using flows like the following.
 
 * Line-item refund flows
-* Test connection functionality when users navigate to **Settings > Setup > Tax > _TaxProviderName_** in an active MSF-enabled BigCommerce control panel
+* Test connection functionality when merchants navigate to **Settings > Setup > Tax > _TaxProviderName_** in an active MSF-enabled BigCommerce control panel
```
Comment (bc-andreadao) [inline docs/api-docs/provider-apis/tax/overview.mdx:150]: Minor question, should this be changed to merchants or **both** merchants and partners? (i.e. do partner stores have the same UI as merchant stores?)

File: docs/api-docs/provider-apis/tax/overview.mdx [hunk 5]
```diff
@@ -164,7 +179,7 @@ Prior to testing a tax provider, the merchant or partner test store should have
...
 * If document submission is supported, navigate to **Settings > Setup > Tax > _TaxProviderName_** and ensure the **Submit Order Data** checkbox is checked
 
-To test the tax provider connection, navigate to **Settings > Setup > Tax > _TaxProviderName_** in an active MSF-enabled BigCommerce control panel and click **Test Connection**. This will trigger a sample tax estimate request to be sent to the estimate URL provided by the tax provider. If the connection is unsuccessful, users can navigate to **Settings > Advanced > Store logs** in an active MSF-enabled BigCommerce control panel to view the error and its description to assist with triaging the issue.
+To test the tax provider connection, navigate to **Settings > Setup > Tax > _TaxProviderName_** in an active MSF-enabled BigCommerce control panel and click **Test Connection**. This will trigger a sample tax estimate request to be sent to the estimate URL provided by the tax provider. If the connection is unsuccessful, merchants can navigate to **Settings > Advanced > Store logs** in an active MSF-enabled BigCommerce control panel to view the error and its description to assist with triaging the issue.
```
Comment (bc-andreadao) [inline docs/api-docs/provider-apis/tax/overview.mdx:182]: Minor question, should this be changed to merchants or **both** merchants and partners? (i.e. do partner stores have the same UI as merchant stores?)

File: docs/api-docs/provider-apis/tax/overview.mdx (hunk 2 above)
Comment (benpratt77) [reply docs/api-docs/provider-apis/tax/overview.mdx:33]: I think merchants is fine, as when a partner is using a store they are acting as a merchant. I.e. the fact it is a partner is irrelevant tot he site itself.

**Review by benpratt77 — APPROVED** [review]

File: docs/api-docs/provider-apis/tax/overview.mdx (hunk 3 above)
Comment (bc-andreadao) [reply docs/api-docs/provider-apis/tax/overview.mdx:124]: Got my answer [here](https://github.com/bigcommerce/dev-docs/pull/1728#discussion_r1152568576)

File: docs/api-docs/provider-apis/tax/overview.mdx (hunk 5 above)
Comment (bc-andreadao) [reply docs/api-docs/provider-apis/tax/overview.mdx:182]: Got my answer [here](https://github.com/bigcommerce/dev-docs/pull/1728#discussion_r1152568576)

File: docs/api-docs/provider-apis/tax/overview.mdx (hunk 4 above)
Comment (bc-andreadao) [reply docs/api-docs/provider-apis/tax/overview.mdx:150]: Got my answer [here](https://github.com/bigcommerce/dev-docs/pull/1728#discussion_r1152568576)

File: docs/api-docs/provider-apis/tax/overview.mdx [hunk 6]
```diff
@@ -30,20 +30,35 @@ Once your tax provider configuration is ready, we'll let you know via email. The
...
+<sup>+</sup> You have the option of supporting a URL that merchants can customize. See [Tax Profile](#tax-profile-optional).
+
+### Tax profile (optional)
+
+BigCommerce sets the tax provider registry with your URLs for [Tax Provider API](/docs/apps-api/tax) requests. BigCommerce sends requests to these URLs, for example, when a shopper requests tax estimates.
+If you would like to support multiple connections for a merchant, BigCommerce can set the registry with flexible URLs that use our new profile option. 
+These URLs would include a reserved string, for example, `https://{profile}/estimate`, allowing merchants to customize the base domain.   
+
+You will need to allow merchants to enter the URL's base domain in the control panel when they connect or install your app. Merchants would also need to be able to self-serve updates to the URL through your app. 
+When you [Establish a Tax Provider connection](#establishing-a-connection), you will include the `profile` base domain along with credentials associated with the merchant's account.
```
Comment (bc-tgomez) [inline docs/api-docs/provider-apis/tax/overview.mdx:56]:
```suggestion
When you [Establish a Tax Provider connection](#establishing-a-connection), include the `profile` base domain along with credentials associated with the merchant's account. 
```

File: docs/api-docs/provider-apis/tax/overview.mdx [hunk 7]
```diff
@@ -104,9 +119,9 @@ The [Get a Connection](/docs/apps-api/tax-app-connection#get-connection-status)
 
 ## Enabling tax providers in the control panel
 
-Once the tax provider's app has been successfully installed and basic authentication credentials have been provided via the Update a Connection request, users will be able to enable the tax provider on all supplied merchant and/or partner test stores provided by the tax provider in [sharing provider details with BigCommerce](#sharing-provider-details-with-bigcommerce).
+Once the tax provider's app has been successfully installed and basic authentication credentials have been provided via the Update a Connection request, merchants will be able to enable the tax provider on all test stores provided by the tax provider in [sharing provider details with BigCommerce](#sharing-provider-details-with-bigcommerce).
```
Comment (bc-tgomez) [inline docs/api-docs/provider-apis/tax/overview.mdx:122]:
```suggestion
Once the tax provider's app is successfully installed and basic authentication credentials are provided through the Update a Connection request, merchants can enable the tax provider on all test stores provided by the tax provider in [sharing provider details with BigCommerce](#sharing-provider-details-with-bigcommerce).
```

**Review by bc-tgomez — APPROVED** [review]
Again, these are quick editorial fixes that you can take or leave (but take the advice about VIA and do a find and replace or something 😸)

File: docs/api-docs/provider-apis/tax/overview.mdx [hunk 8]
```diff
@@ -6,20 +6,20 @@ The [Tax Provider API](/docs/apps-api/tax) allows you to provide business-to-con
...
+To get your app ID, start creating an app in the [Developer Portal](https://devtools.bigcommerce.com/), and fill out the information on [Step 2 Technical](/api-docs/apps/guide/publishing#add-technical-information). In the URL, you will see your unique app ID.
 
 [Learn more about finding your app's ID](/api-docs/apps/tutorials/id).
 
 ## Sharing provider details with BigCommerce
 
-Once you've obtained your app ID, please share your app ID as well as the below information with BigCommerce by emailing [taxproviderapi@bigcommerce.com](mailto:taxproviderapi@bigcommerce.com). This allows us to create your tax provider configuration, which will take approximately 5 days.
+Once you've obtained your app ID, share your app ID as well as the below information with BigCommerce by emailing [taxproviderapi@bigcommerce.com](mailto:taxproviderapi@bigcommerce.com). This allows us to create your tax provider configuration, which will take approximately five days.
```
Comment (bc-traciporter) [inline docs/api-docs/provider-apis/tax/overview.mdx:20]:
```suggestion
Once you've obtained your app ID, share your app ID and the information below with BigCommerce by emailing [taxproviderapi@bigcommerce.com](mailto:taxproviderapi@bigcommerce.com). This information allows us to create your tax provider configuration, which will take approximately five days.
```

File: docs/api-docs/provider-apis/tax/overview.mdx [hunk 9]
```diff
@@ -6,20 +6,20 @@ The [Tax Provider API](/docs/apps-api/tax) allows you to provide business-to-con
...
 
-Once your tax provider configuration is ready, we'll let you know via email. The email will also include your `provider_id` which is required when [establishing a connection](#establishing-a-connection) with the [Tax Provider API](/docs/apps-api/tax).
+Once your tax provider configuration is ready, we'll let you know through email. The email will also include your `provider_id` which is required when [establishing a connection](#establishing-a-connection) with the [Tax Provider API](/docs/apps-api/tax).
```
Comment (bc-traciporter) [inline docs/api-docs/provider-apis/tax/overview.mdx:22]:
```suggestion
Once your tax provider configuration is ready, we'll let you know through email. The email will also include your `provider_id`, which is required when [establishing a connection](#establishing-a-connection) with the [Tax Provider API](/docs/apps-api/tax).
```

File: docs/api-docs/provider-apis/tax/overview.mdx [hunk 10]
```diff
@@ -30,20 +30,35 @@ Once your tax provider configuration is ready, we'll let you know via email. The
...
 
+<sup>+</sup> You have the option of supporting a URL that merchants can customize. See [Tax Profile](#tax-profile-optional).
+
+### Tax profile (optional)
+
+BigCommerce sets the tax provider registry with your URLs for [Tax Provider API](/docs/apps-api/tax) requests. BigCommerce sends requests to these URLs, for example, when a shopper requests tax estimates.
+If you would like to support multiple connections for a merchant, BigCommerce can set the registry with flexible URLs that use our new profile option. 
+These URLs would include a reserved string, for example, `https://{profile}/estimate`, allowing merchants to customize the base domain.   
+
+You will need to allow merchants to enter the URL's base domain in the control panel when they connect or install your app. Merchants would also need to be able to self-serve updates to the URL through your app.
```
Comment (bc-traciporter) [inline docs/api-docs/provider-apis/tax/overview.mdx:55]:
```suggestion
When connecting or installing your app, you must allow merchants to enter the URL's base domain in the control panel. Merchants need to be able to self-serve updates to the URL through your app. 
```

File: docs/api-docs/provider-apis/tax/overview.mdx [hunk 11]
```diff
@@ -30,20 +30,35 @@ Once your tax provider configuration is ready, we'll let you know via email. The
...
+
+BigCommerce sets the tax provider registry with your URLs for [Tax Provider API](/docs/apps-api/tax) requests. BigCommerce sends requests to these URLs, for example, when a shopper requests tax estimates.
+If you would like to support multiple connections for a merchant, BigCommerce can set the registry with flexible URLs that use our new profile option. 
+These URLs would include a reserved string, for example, `https://{profile}/estimate`, allowing merchants to customize the base domain.   
+
+You will need to allow merchants to enter the URL's base domain in the control panel when they connect or install your app. Merchants would also need to be able to self-serve updates to the URL through your app. 
+When you [Establish a Tax Provider connection](#establishing-a-connection), include the `profile` base domain along with credentials associated with the merchant's account. 
+
+<Callout type="info">
+  You must liaise with the merchant to provide them with customized URLs. BigCommerce merely consumes the URL values that the merchant enters and uses the provided string value for Tax Provider API requests.
```
Comment (bc-traciporter) [inline docs/api-docs/provider-apis/tax/overview.mdx:59]:
```suggestion
  You must liaise with the merchant to provide them with customized URLs. BigCommerce merely consumes the URL values the merchant enters and uses the provided string value for Tax Provider API requests.
```

File: docs/api-docs/provider-apis/tax/overview.mdx [hunk 12]
```diff
@@ -63,7 +78,7 @@ Additionally, when providing the details for a sandbox tax provider configuratio
 
 ## Building the app
 
-Tax providers are required to build a BigCommerce [single-click app](/api-docs/apps/guide/types#single-click) in order to utilise the Tax Provider API to provide tax estimates and submit tax documents. A BigCommerce single-click app provides many benefits, for example, it enables tax providers to promote their solution in the BigCommerce App Marketplace, ask for merchant authorisation of API scopes during app install, as well as enable configuration of tax provider settings via an iFrame in the BigCommerce control panel.
+Tax providers are required to build a BigCommerce [single-click app](/api-docs/apps/guide/types#single-click) in order to utilise the Tax Provider API to provide tax estimates and submit tax documents. A BigCommerce single-click app provides many benefits, for example, it enables tax providers to promote their solution in the BigCommerce App Marketplace, ask for merchant authorisation of API scopes during app install, as well as enable configuration of tax provider settings through an iFrame in the BigCommerce control panel.
```
Comment (bc-traciporter) [inline docs/api-docs/provider-apis/tax/overview.mdx:81]:
```suggestion
Tax providers must build a BigCommerce [single-click app](/api-docs/apps/guide/types#single-click) to utilize the Tax Provider API to provide tax estimates and submit tax documents. A BigCommerce single-click app provides many benefits. For example, it enables tax providers to promote their solution in the BigCommerce App Marketplace, ask for merchant authorization of API scopes during app install, and allow configuration of tax provider settings through an iFrame in the BigCommerce control panel.
```

File: docs/api-docs/provider-apis/tax/overview.mdx [hunk 13]
```diff
@@ -104,9 +119,9 @@ The [Get a Connection](/docs/apps-api/tax-app-connection#get-connection-status)
 
 ## Enabling tax providers in the control panel
 
-Once the tax provider's app has been successfully installed and basic authentication credentials have been provided via the Update a Connection request, users will be able to enable the tax provider on all supplied merchant and/or partner test stores provided by the tax provider in [sharing provider details with BigCommerce](#sharing-provider-details-with-bigcommerce).
+Once the tax provider's app is successfully installed and basic authentication credentials are provided through the Update a Connection request, merchants can enable the tax provider on all test stores provided by the tax provider in [sharing provider details with BigCommerce](#sharing-provider-details-with-bigcommerce).
```
Comment (bc-traciporter) [inline docs/api-docs/provider-apis/tax/overview.mdx:122]:
```suggestion
Once you have successfully installed the tax provider's app and provided basic authentication credentials through the Update a Connection request, merchants can enable the tax provider on all test stores supplied by the tax provider in [sharing provider details with BigCommerce](#sharing-provider-details-with-bigcommerce).
```

File: docs/api-docs/provider-apis/tax/overview.mdx [hunk 14]
```diff
@@ -132,7 +147,7 @@ Estimate requests are not expected after the following events.
 Estimate calls will also be made in the control panel when using flows like the following.
```
Comment (bc-traciporter) [inline docs/api-docs/provider-apis/tax/overview.mdx:147]:
```suggestion
Estimate calls will also be made in the control panel when using flows like the following:
```

File: docs/api-docs/provider-apis/tax/overview.mdx (hunk 5 above)
Comment (bc-traciporter) [inline docs/api-docs/provider-apis/tax/overview.mdx:182]:
```suggestion
To test the tax provider connection, navigate to **Settings > Setup > Tax > _TaxProviderName_** in an active MSF-enabled BigCommerce control panel and click **Test Connection**. This selection will trigger a sample tax estimate request to be sent to the estimate URL provided by the tax provider. If the connection is unsuccessful, merchants can navigate to **Settings > Advanced > Store logs** in an active MSF-enabled BigCommerce control panel to view the error and its description to assist with triaging the issue.
```

**Review by bc-traciporter — CHANGES_REQUESTED** [review]
I suggested some changes.

**Review by bc-traciporter — APPROVED** [review]
LGTM!

**Pull request merged**